core/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardRepository.java

```diff
@@ -204,7 +204,9 @@ public void verify(String seed) {
                 throw new RepositoryVerificationException(repositoryName, "store location [" + blobStore + "] is not accessible on the node [" + localNode + "]", exp);
             }
         } else {
-            throw new RepositoryVerificationException(repositoryName, "store location [" + blobStore + "] is not shared between node [" + localNode + "] and the master node");
+            throw new RepositoryVerificationException(repositoryName, "a file written by master to the store [" + blobStore + "] cannot be accessed on the node [" + localNode + "]. "
+                    + "This might indicate that the store [" + blobStore + "] is not shared between this node and the master node or "
+                    + "that permissions on the store don't allow reading just created files");
```

Comment:
What I meant here is that it couldn't read the file that was just created by the master. I will change the language. Thanks!

We now log this. See #11763, but there is really no more information to log here because there is no way to distinguish between file doesn't exist and file exists but you don't have permissions to read it on this level.